source/game_sa/Camera.cpp

```diff
@@ -867,7 +867,25 @@ void CCamera::CameraPedAimModeSpecialCases(CPed* ped) {
 
 // 0x50CDE0
 void CCamera::CameraVehicleModeSpecialCases(CVehicle* vehicle) {
-    plugin::CallMethod<0x50CDE0, CCamera*, CVehicle*>(this, vehicle);
+    CCollision::ms_bCamCollideWithObjects = false;
+
+    float speed = (vehicle->m_vecMoveSpeed).Magnitude();
+
+    if (speed <= 0.2f) {
+        CCollision::ms_relVelCamCollisionVehiclesSqr = 0.1f;
+        CCollision::ms_bCamCollideWithVehicles = true;
+        CCollision::ms_bCamCollideWithPeds = true;
+        CCollision::ms_bCamCollideWithObjects = true;
+    } else {
+        CCollision::ms_relVelCamCollisionVehiclesSqr = 1.0f;
+        CCollision::ms_bCamCollideWithVehicles = true;
+        CCollision::ms_bCamCollideWithPeds = false;
+        CCollision::ms_bCamCollideWithObjects = false;
+    }
```

Comment:
Nit-pick. I'll leave this one up to you.
```suggestion
    const auto slow = speed <= 0.2f;
    CCollision::ms_bCamCollideWithVehicles = true;
    CCollision::ms_bCamCollideWithPeds = slow;
    CCollision::ms_bCamCollideWithObjects = slow;
    CCollision::ms_relVelCamCollisionVehiclesSqr = slow ? 0.1f : 1.0f;
```